nvflare/private/fed/server/server_engine.py

```diff
@@ -570,101 +477,50 @@ def send_aux_request(self, targets: [], topic: str, request: Shareable, timeout:
 
     def sync_clients_from_main_process(self):
         with self.parent_conn_lock:
-            data = {ServerCommandKey.COMMAND: ServerCommandNames.GET_CLIENTS, ServerCommandKey.DATA: {}}
-            self.parent_conn.send(data)
-            return_data = self.parent_conn.recv()
+            return_data = self._get_clients_data(self.args.job_id)
             clients = return_data.get(ServerCommandKey.CLIENTS)
             self.client_manager.clients = clients
 
-    def parent_aux_send(self, targets: [], topic: str, request: Shareable, timeout: float, fl_ctx: FLContext) -> dict:
-        with self.parent_conn_lock:
-            data = {
-                ServerCommandKey.COMMAND: ServerCommandNames.AUX_SEND,
-                ServerCommandKey.DATA: {
-                    "targets": targets,
-                    "topic": topic,
-                    "request": request,
-                    "timeout": timeout,
-                    "fl_ctx": get_serializable_data(fl_ctx),
-                },
-            }
-            self.parent_conn.send(data)
-            return_data = self.parent_conn.recv()
-            return return_data
+    def _get_clients_data(self, job_id):
+        request = new_cell_message({CellMessageHeaderKeys.JOB_ID: job_id}, fobs.dumps({}))
+        return_data = self.server.cell.send_request(
+            target=FQCN.ROOT_SERVER,
+            channel=CellChannel.SERVER_PARENT_LISTENER,
+            topic=ServerCommandNames.GET_CLIENTS,
+            request=request,
+        )
+        data = fobs.loads(return_data.payload)
+        if data.get(ServerCommandKey.CLIENTS):
+            return data
+        else:
+            time.sleep(1.0)
+            return self._get_clients_data(job_id)
```

Comment:
If for whatever reason we can't get the clients from the return_data's payload,
this function never ends?

Reply:
Added timeout control.